packages/observability-node/src/logger.ts

```diff
@@ -0,0 +1,89 @@
+import winston from "winston";
+import LokiTransport from "winston-loki";
+import dbg from "debug";
+
+const GRAFANA_USER = "909360";
+const GRAFANA_PUBLIC_TOKEN =
+  "glc_eyJvIjoiOTEyOTQzIiwibiI6IndyaXRlLW90ZWwtcmVwbGF5LWNsaSIsImsiOiJ0UnFsOXV1a2QyQUI2NzIybDEzSkRuNDkiLCJtIjp7InIiOiJwcm9kLXVzLWVhc3QtMCJ9fQ=="; // write-only permissions.
+const GRAFANA_BASIC_AUTH = `${GRAFANA_USER}:${GRAFANA_PUBLIC_TOKEN}`;
+const HOST = "https://logs-prod-006.grafana.net";
+
+type LogLevel = "error" | "warn" | "info" | "debug";
+
+type Tags = Record<string, unknown>;
+
+class Logger {
+  private userOrWorkspaceId?: string;
+  private grafana: {
+    logger: winston.Logger;
+    flush: () => Promise<null>;
+  };
+
+  private name: string;
+  private localDebugger: debug.Debugger;
+
+  constructor(name: string) {
+    this.name = `replayio:${name}`;
+    this.localDebugger = dbg(name);
+    this.grafana = this.initGrafana();
+  }
+
+  private initGrafana() {
+    const lokiTransport = new LokiTransport({
+      host: HOST,
+      labels: { app: this.name },
+      json: true,
+      basicAuth: GRAFANA_BASIC_AUTH,
+      format: winston.format.json(),
+      replaceTimestamp: true,
+      onConnectionError: err => this.localDebugger("Grafana connection error %s", err),
+      gracefulShutdown: true,
+    });
+
+    return {
+      logger: winston.createLogger({
+        // Levels greater than or equal to "info" ("info", "warn", "error") will be logged.
+        // See https://github.com/winstonjs/winston?tab=readme-ov-file#logging.
+        level: "info",
+        transports: [lokiTransport],
+      }),
+      flush: async () => lokiTransport?.flush(),
+    };
+  }
+
+  identify(userOrWorkspaceId: string) {
+    this.userOrWorkspaceId = userOrWorkspaceId;
+  }
+
+  private log(message: string, level: LogLevel, tags?: Tags) {
```

Comment:
What's your longer term thinking here, once we add an additional telemetry endpoint (like Mixpanel) of we have only one method (`log`) that does local+remote

Reply:
Sorry, I'm not sure if I understand what you are asking.

Reply:
Sorry! I was thinking, for example, that `replayio` uses Mixpanel to log data _and_ it has the concept of `debug` for (local only) debug logging. So if we updated that package to use this new shared observability code, would it...

1. Replace its current debugger with this one _and_ continue logging one-off events to Mixpanel?
2. Use this logger for both Grafana _and_ Mixpanel?

Assuming we are intending for an "observability" package to include things like Mixpanel too, so we'd go with the second option, then I'm wondering what that API might look like? Would `log` remain Grafana-only and we'd add some new method(s) for Mixpanel?

Reply:
Good question.

"Replace its current debugger with this one and continue logging one-off events to Mixpanel?"

I think we should eventually replace all debugging with this logger since it handles both local and Grafana logging. I'll address Mixpanel below.

"Use this logger for both Grafana and Mixpanel?"

In my mind, Mixpanel is conceptually separate from logging. Mixpanel is for tracking user interactions for analytical purposes, while logging is captures more verbose info about the system's state and behavior.

I would prefer there to be separate component for each aspect of observability (Honeycomb, Mixpanel, logging). In the backend, we use context objects to unify components of our observability (Honeycomb, logging, and Sentry).

The context object allows us to pass common attributes for a session, user, and recording around. It has a logger on it. We log exceptions to Sentry through the `Context.Logger` object. We have a separate honeycomb span-creator function that accepts contexts which it uses both to establish trace/span hierarchy and to pass common attributes to the traces.  We could do something similar in the future here too.

I'm open to suggestions though, and I'm curious to hear what your thoughts are.

Reply:
> In my mind, Mixpanel is conceptually separate from logging.

Ah, yes. This distinction makes sense. I was mixing up "observability" (which Mixpanel does fall under) and "logging" which it doesn't. 👍🏼 

I don't think I can picture what you're describing yet but I'm interested to learn more about it. (Happy to talk about it too if you ever want to.)

Reply:
Definitely, I'm happy to talk about it.

I think logging and Mixpanel are both "observability," but I also thought "telemetry" and "observability" were the same thing all of last week and I don't think that is right either 😭so I may not be the best person to ask.

To me, anything that helps us understand the behavior of our application, whether locally or remotely, is "observability".